booster/library/Booster/Util.hs

```diff
@@ -155,3 +159,21 @@ withFastLogger mFormattedTime (Just fp) log' =
         handleExists e
             | isDoesNotExistError e = return ()
             | otherwise = throwIO e
+
+{- |  Make 'IO' action which get cached formatted local time.
+Use this to avoid the cost of frequently time formatting by caching an
+auto updating formatted time, this cache update every 1 second.
+more detail in "Control.AutoUpdate"
+
+
+Borrowed almost verbatim from the fast-logger package: https://hackage.haskell.org/package/fast-logger-3.2.3/docs/src/System.Log.FastLogger.Date.html#newTimeCache, but the timestamp resolution and the action to get and format the time are tweaked
+-}
+newTimeCache :: TimeFormat -> IO (IO FormattedTime)
+newTimeCache fmt =
+    mkAutoUpdate
+        defaultUpdateSettings{updateFreq = 100}
+            { updateAction = formatSystemTime fmt <$> getSystemTime
+            }
+
+formatSystemTime :: TimeFormat -> SystemTime -> ByteString
+formatSystemTime fmt = BS.pack . formatTime defaultTimeLocale (BS.unpack fmt) . systemToUTCTime
```

Comment:
(nit) Adjust the comment at the top which says "every 1 second".
Should we maybe hard-wire the time format here?

Reply:
I've updated the comment and hardwired the format string. We can also switch the formatting off now and just the raw nanoseconds. This will be useful when we parse the JSON logs for post-processing.